Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,253 @@
+        logging.info(f'There is a lock file for job id: {job_id_of_the_existing_lock}')
+        job_id_of_the_existing_lock_status = check_job_status(gitlab_status_token, job_id_of_the_existing_lock)
+        logging.info(f'the status of job id: {job_id_of_the_existing_lock} is: {job_id_of_the_existing_lock_status}')
+        if job_id_of_the_existing_lock_status != 'running':
+            # machine found! removing the not relevant lock and create a now one
+            remove_machine_lock_file(storage_bucket, gcs_locks_path, machine, job_id_of_the_existing_lock)
+            lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
+            lock_machine_name = machine
+    else:
+        # machine found! create lock file
+        logging.info('There is no a lock file')
+        lock_machine(storage_bucket, gcs_locks_path, machine, job_id)
+        lock_machine_name = machine
+    return lock_machine_name
+
+
+def main():
+    install_logging('lock_cloud_machines.log', logger=logging)
+    logging.info('Starting search a CLOUD machine to lock')
+    options = options_handler()
+    storage_client = storage.Client.from_service_account_json(options.service_account)
+    storage_bucket = storage_client.bucket('xsoar-ci-artifacts')
+    number_machines_to_lock = int(options.number_machines_to_lock)
+
+    logging.info('adding job_id to the queue')
+    adding_build_to_the_queue(storage_bucket, options.gcs_locks_path, options.ci_job_id)
+
+    # running until the build is the first in queue
+    first_in_the_queue = False
+    while not first_in_the_queue:
+        my_place_in_the_queue, previous_build = get_my_place_in_the_queue(storage_client, options.gcs_locks_path,
+                                                                          options.ci_job_id)
+        logging.info(f'my place in the queue is: {my_place_in_the_queue}')
+
+        if my_place_in_the_queue == 0:
+            first_in_the_queue = True
+        else:
+            # we check the status of the build that is ahead of me in the queue
+            previous_build_status = check_job_status(options.gitlab_status_token, previous_build)
+            if previous_build_status != 'running':
+                # delete the lock file of the build because its not running
+                remove_build_from_queue(storage_bucket, options.gcs_locks_path, previous_build)
+            else:
+                sleep(random.randint(8, 13))
+
+    logging.info('Start searching for an empty machine')
+    if options.lock_machine_name:
+        logging.info('trying to lock the given machine: {options.lock_machine_name}')
+        list_machines = [options.lock_machine_name]
+    else:
+        logging.info('gets all machines names')
+        test_machines_list = storage_bucket.blob(f'{options.gcs_locks_path}/{options.test_machines_list}')
+        list_machines = test_machines_list.download_as_string().decode("utf-8").split()
+        random.shuffle(list_machines)
```

Comment:
I think this code block should be extracted to an additional function: `create_list_of_machines_to_run()` 
such that :
```python
list_machines = create_list_of_machines_to_run()
```